src/main/java/uk/gov/digital/ho/hocs/client/info/InfoClient.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.digital.ho.hocs.client.info;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import uk.gov.digital.ho.hocs.application.RestClient;
+import uk.gov.digital.ho.hocs.client.info.dto.GetTopicsResponse;
+import uk.gov.digital.ho.hocs.client.info.dto.Topic;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+@Slf4j
+@Component
+public class InfoClient {
+
+    private final RestClient restClient;
+    private final String serviceBaseURL;
+
+    public InfoClient(RestClient restClient, @Value("${case.creator.info-service}") String serviceBaseURL) {
+        this.restClient = restClient;
+        this.serviceBaseURL = serviceBaseURL;
+    }
+
+    public List<Topic> getTopicsForParent(String messageId, UUID parentTopicUUID) {
```

Comment:
Could this be cached, we'd be doing this call for every case with a topic?

Reply:
The resulting Map<String, UUID> is cached. Tested here https://github.com/UKHomeOffice/hocs-case-creator/blob/feature/HOCS-6385-create-primary-topic/src/test/java/uk/gov/digital/ho/hocs/domain/queue/migration/TopicMapperTest.java#L66

Reply:
Ah I see it here: `getTopicMapping` has a null check.